IOS/SwiftSeedProject/SwiftSeedProject/Application/Service/BaseService.swift

```diff
@@ -7,25 +7,34 @@
 //
 
 import Foundation
+import SwiftyJSON
 
 class BaseService<T: PersistenceObject>  {
     
-    private let persistence: Persistence
-    private let restClient: AnyObject? = nil
-    
-    //TODO: we should inclued the API access also
-    init(persistence: Persistence) {
+    private let persistence: Persistence!
+    internal let api: Api!
+
+    init(persistence: Persistence, api: Api) {
         self.persistence = persistence
+        self.api = api
     }
     
     public func getAll() -> [T] {
         return persistence.getAll(entityName: T.EntityName)
     }
     
-    public func getEntityBy(identifier: Int) -> T {
+    public func getEntityBy(identifier: String) -> T? {
         return persistence.getBy(entityIdentifier: identifier, entityName: T.EntityName)
     }
     
+    public func createEntity(json: JSON) {
```

Comment:
I think that the Service layer should not be able to save from json. Maybe the api should has some way to save on core data.